src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -608,17 +628,14 @@ private Key CreateKey(string name, string secret)
             return _keyValueConverterFactory.WriteKey(key);
         }
 
-        internal static string GenerateSecret()
+        internal static string GenerateIdentifiableSecret(ulong seed)
         {
-            using (var rng = RandomNumberGenerator.Create())
-            {
-                byte[] data = new byte[40];
-                rng.GetBytes(data);
-                string secret = Convert.ToBase64String(data);
+            string secret = IdentifiableSecrets.GenerateIdentifiableKey(seed, 40, AzureFunctionsSignature);
```

Comment:
>GenerateIdentifiableKey [](http://example.com/codeflow?start=48&length=23)

This is the essential change. This helper will produce a base64-encoded key, seven bytes of which are dedicated to a fixed signature (when encoded) and a checksum. The remaining entropy of the key is sufficient to meet security standards. Alternately, we could increase the size of the generated key in order to preserve full 320 bits of entropy (as exists today).